app/src/main/java/com/youssef/fixit/Auth/LoginFragment/LoginPresenter.java

```diff
@@ -112,4 +113,9 @@ public void onFailure(Call<CreateBid> call, Throwable t) {
             }
         });
     }
+
+    void onDestroy() {
+        view = null;
+        iSharedPreference=null;
```

Comment:
If you release the shared preference here it won't be singleton. The instance is released and you will be creating a new one single instance every time. no need to remove it here I would say.
Also, good that you renamed the interface to `iSharedPreference` but no need to name its instance iSharedPreference it's better to name it `sharedPreference` because the presenter actually assuming that the passed dependency is an implementation not an interface. The presenter does accept interface but this to provide more inversions.